matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/DefaultCryptoService.kt

```diff
@@ -427,7 +429,10 @@ internal class DefaultCryptoService @Inject constructor(
                 if (isStarted()) {
                     // Make sure we process to-device messages before generating new one-time-keys #2782
                     deviceListManager.refreshOutdatedDeviceLists()
-                    oneTimeKeysUploader.maybeUploadOneTimeKeys()
+                    // The presence of device_unused_fallback_key_types indicates that the server supports fallback keys.
+                    // If there's no unused signed_curve25519 fallback key we need a new one.
+                    val shouldGenerateFallbackKey = !syncResponse.deviceUnusedFallbackKeyTypes?.contains(KEY_SIGNED_CURVE_25519_TYPE).orTrue()
```

Comment:
I think it is equivalent to the simpler version:

`val shouldGenerateFallbackKey = syncResponse.deviceUnusedFallbackKeyTypes?.contains(KEY_SIGNED_CURVE_25519_TYPE).orFalse()`

Remove `!` and use `orFalse`. WDYT?

> The presence of device_unused_fallback_key_types indicates that the server supports fallback keys.

Also to ensure that there is no mistake, and for clarity, you should probably cut this line in 2 tests, i.e.:

```kotlin
val shouldGenerateFallbackKey = if (syncResponse.deviceUnusedFallbackKeyTypes != null) {
    if(syncResponse.deviceUnusedFallbackKeyTypes.contains(KEY_SIGNED_CURVE_25519_TYPE) {
        // Server already have the fallback key
        false
    } else {
        // Generate the fallback key
        true
    }
} else {
    // Server does not support fallbackKey
    false
}
```

Reply:
Your first suggestion is not the same. Because we shouldn't generate a key if it is already in the list.

Reply:
Yes, you are right. So for clarity, please use my second suggestion, to avoid 🤯

Reply:
Lint is not happy with the clear one of (redundant) if/else. I prefer a happy medium:

```
val shouldGenerateFallbackKey = if (syncResponse.deviceUnusedFallbackKeyTypes != null) {
    // Generate a fallback key only if the server does not already have an unused fallback key.
    !syncResponse.deviceUnusedFallbackKeyTypes.contains(KEY_SIGNED_CURVE_25519_TYPE)
} else {
    // Server does not support fallbackKey
    false
}
```

Reply:
Yes, using if/else just for comments was a bit ugly :)